File: solutions/go/spiral-matrix/1/spiral_matrix.go
package spiralmatrix

func SpiralMatrix(size int) [][]int {
	result := [][]int{}
    for row := 0; row < size; row++ {
        result = append(result, make([]int, size))
    }

    row, col := 0, 0
    r, c := 0, 1
    for i := 1; i <= size*size; i++ {
        result[row][col] = i
        if row+r < 0 || size <= row+r || col+c < 0 ||
           size <= col+c || result[row+r][col+c] != 0 {
               r, c = c, -r
           }
        row, col = row+r, col+c
    }
    return result
}

File: solutions/vbnet/spiral-matrix/1/SpiralMatrix.vb
Imports System

Public Class SpiralMatrix
    Public Shared Function GetMatrix(ByVal size As Integer) As Integer(,)
        Dim result(size-1, size-1) As Integer
        Dim row = -1, col = 0, r = 1, c = 0
        For i = 1 To size * size
          If row+r < 0 OrElse row+r >= size OrElse col+c >= size OrElse result(col+c, row+r) > 0 Then
            Dim t = -c
            c = r
            r = t
          End If

          row = row+r
          col = col+c
          result(col, row) = i
      Next
      Return result
    End Function
End Class
